src/ulid.cr

```diff
@@ -19,6 +19,23 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # is ulid the correct size AND only contain valid characters?
+    ulid.size == TIME_LEN + RANDOM_LEN && (ulid =~ /[^0123456789ABCDEFGHJKMNPQRSTVWXYZ]/).nil?
```

Comment:
```suggestion
    # Incorrect length
    return false unless ulid.size == TIME_LEN + RANDOM_LEN

    # Invalid chars
    return false unless ulid.chars.all? &.in?(ENCODING)

    return true
```

Reply:
@dukeraphaelng @caspiano had this originally, is there a preferred approach?